### DISCUSSION TITLE
Securing custom routes for app and device services

### DISCUSSION THREAD

**Discussion by jrtitus** [edgexfoundry/community#143] [Q&A] [answered]
Since v3.0 supports the use of a JWT for communicating with the EdgeX service REST APIs, is there a plan for integrating that feature with the [custom endpoints](https://docs.edgexfoundry.org/3.0/microservices/application/AdvancedTopics/#custom-rest-endpoints) functionality from the Application and Device Service SDKs?

**Comment by bnevis-i** [comment]
Probably best answered by @lenny-intel, but I think it should be as simple as wrapping your handler with `authenticationHook()`

https://github.com/edgexfoundry/edgex-go/blob/main/internal/core/data/router.go#L36

**Reply by jrtitus** [reply]
Thanks, @bnevis-i. The problem I see from here is it seems like there is no access to `dic.Get` (or something equivalent) when `AddRoute` is being called within the `main` function of an application service... or at least I can't find the right way to access it.

https://github.com/edgexfoundry/edgex-go/blob/main/internal/core/data/router.go#L29-L30

**Comment by lenny-goodell** [comment]
@bnevis-i , this was missed in the Microservice Auth implementation. None of that App SDK  end points are secured. Still checking the Go Device SDK .

**Reply by lenny-goodell** [reply, marked as answer]
The internal Device SDK routes where handled, but not the ones via AddRoute here: https://github.com/edgexfoundry/device-sdk-go/blob/main/internal/controller/http/restrouter.go#L87

@jrtitus , we'll add issues to each SDK to ensure all routes are secured when running in secure mode.

**Reply by jrtitus** [reply]
Ah! Okay, thanks @lenny-intel

**Reply by lenny-goodell** [reply]
Issues created and will be address in the upcoming Napa release.
